Mathlib/Data/Finset/Sort.lean

```diff
@@ -335,3 +335,60 @@ lemma nonempty_orderEmbedding_of_finite_infinite
   haveI := Fintype.ofFinite α
   obtain ⟨s, hs⟩ := Infinite.exists_subset_card_eq β (Fintype.card α)
   exact ⟨((Fintype.orderIsoFinOfCardEq α rfl).symm.toOrderEmbedding).trans (s.orderEmbOfFin hs)⟩
+
+section OrderedTuple
+
+variable {α : Type*} [LinearOrder α] {t : ℕ}
+
+/-- For a linearly ordered type α, `Fin t ↪o α` is equivalent to the type of `t`-sets. -/
+@[simps] def tupleEquivFinset : (Fin t ↪o α) ≃ {s : Finset α // s.card = t} where
+  toFun f := ⟨Finset.univ.map f.toEmbedding, by simp⟩
+  invFun s := Finset.orderEmbOfFin s.1 s.2
+  left_inv _ := Eq.symm <| Finset.orderEmbOfFin_unique' _ (by simp)
+  right_inv := by
+    rintro ⟨s, rfl⟩
+    simp only [Finset.map_eq_image, RelEmbedding.coe_toEmbedding, Subtype.mk.injEq]
+    rw [← Finset.coe_inj, ← Finset.range_orderEmbOfFin s rfl]
+    simp
+
+/-- Add a new element greater than all existing elements to the end of a tuple. -/
+def appendRight (f : Fin t ↪o α) (a : α) (ha : ∀ i, f i < a) : Fin (t + 1) ↪o α :=
+  OrderEmbedding.ofStrictMono (Fin.lastCases a f) (
+    Fin.lastCases (fun b h ↦ (h.not_ge <| Fin.le_last b).elim)
+      (fun i ↦ Fin.lastCases (by simp [ha]) fun _ hij ↦ by simpa using hij))
```

Comment:
Should this really be in `Data.Finset.Sort`? I would rather expect it around `Order.Fin.Tuple`